sdk/storage/Azure.Storage.Blobs/src/Models/BlockListOptions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Specifies whether to return the list of committed blocks, the list of uncommitted blocks, or both lists together.
+    /// </summary>
+    [Flags]
+    public enum BlockListOptions
+    {
+        /// <summary>
+        /// Flag to specify returning both committed and uncommitted blocks.
+        /// </summary>
+        All = Committed | Uncommitted,
+
+        /// <summary>
+        /// Flag to specify returning the list of committed blocks.
+        /// </summary>
+        Committed = 1,
+
+        /// <summary>
+        /// Flag to specify returning the list of uncommitted blocks.
+        /// </summary>
+        Uncommitted = 2
+    }
+}
+
+namespace Azure.Storage.Blobs
+{
+    /// <summary>
+    /// BlockListOptions extensions
+    /// </summary>
+    internal static partial class BlobExtensions
+    {
+        /// <summary>
+        /// Convert the BlockListOptions into a BlockListType.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns>The BlockList response.</returns>
+        internal static BlockListType ToBlockListType(this BlockListOptions options)
+        {
+            if (options.HasFlag(BlockListOptions.All) || options == default)
```

Comment:
`HasFlag` is kind of slow because it boxes.  `(options & BlockListOptions.All) == BlockListOptions.All` is probably better for these.

Reply:
Should we default everything else to `All`?  If I pass `(BlockListOptions)7`, I think `All` is a better choice than `Uncommitted`.